test/env_test.rb

```diff
@@ -72,11 +72,6 @@ def test_request_create_stores_ssl_options
     assert_equal false, env.ssl.verify
   end
 
-  def test_request_create_stores_proxy_options
-    env = make_env
-    assert_equal 'proxy.com', env.request.proxy.host
-  end
```

Comment:
This test was removed because of where (and how) the proxy options are merged into the request object now.  The `make_env` method apart of this test class calls `connection.build_request` to build the request.  This PR changes the code so that t[he proxy options are merged later](https://github.com/somethingnew2-0/faraday/blob/6bc251833b5c837825aaa4055b19271d4d212d47/lib/faraday/connection.rb#L379) in the `connection.run_request` method instead because we need the destination url in order to compare to the `no_proxy` domain list.

```
def make_env(method = :get, connection = @conn, &block)
  request = connection.build_request(method, &block)
  request.to_env(connection)
end
```